controllers/dbuser_controller.go

```diff
@@ -0,0 +1,344 @@
+	if err != nil {
+		// failed to parse database admin secret
+		return r.manageError(ctx, dbucr, err, false)
+	}
+
+	dbuser.AccessType = dbucr.Spec.AccessType
+	dbuser.Password = creds.Password
+	dbuser.Username = fmt.Sprintf("%s-%s", dbucr.GetObjectMeta().GetNamespace(), dbucr.GetObjectMeta().GetName())
+
+	if dbucr.GetDeletionTimestamp() != nil {
+		dbucr.Status.Phase = DBUCR_PHASE_DELETING
+		if containsString(dbucr.ObjectMeta.Finalizers, "dbuser."+dbucr.Name) {
+			if err := database.DeleteUser(db, dbuser, adminCred); err != nil {
+				logrus.Errorf("DBUser: namespace=%s, name=%s failed deleting a user - %s", dbucr.Namespace, dbucr.Name, err)
+				return r.manageError(ctx, dbucr, err, false)
+			}
+			kci.RemoveFinalizer(&dbucr.ObjectMeta, "dbuser."+dbucr.Name)
+			err = r.Update(ctx, dbucr)
+			if err != nil {
+				logrus.Errorf("error resource updating - %s", err)
+				return r.manageError(ctx, dbucr, err, false)
+			}
+			kci.RemoveFinalizer(&dbcr.ObjectMeta, "dbuser."+dbucr.Name)
+			err = r.Update(ctx, dbcr)
+			if err != nil {
+				logrus.Errorf("error resource updating - %s", err)
+				return r.manageError(ctx, dbucr, err, false)
+			}
+
+		}
+	} else {
+		if !dbcr.Status.Status {
+			err := fmt.Errorf("database %s is not ready yet", dbcr.Name)
+			return r.manageError(ctx, dbucr, err, true)
+		}
+
+		//Init the DbUser struct depending on a type
+		if !dbucr.Status.Status {
+			if !dbucr.Status.Created {
+				dbucr.Status.Phase = DBUCR_PHASE_CREATING
+				r.Log.Info(fmt.Sprintf("creating a user: %s", dbucr.GetObjectMeta().GetName()))
+				if err := database.CreateUser(db, dbuser, adminCred); err != nil {
+					return r.manageError(ctx, dbucr, err, false)
+				}
+				kci.AddFinalizer(&dbucr.ObjectMeta, "dbuser."+dbucr.Name)
+				err = r.Update(ctx, dbucr)
+				if err != nil {
+					logrus.Errorf("error resource updating - %s", err)
+					return r.manageError(ctx, dbucr, err, false)
+				}
+				kci.AddFinalizer(&dbcr.ObjectMeta, "dbuser."+dbucr.Name)
+				err = r.Update(ctx, dbcr)
+				if err != nil {
+					logrus.Errorf("error resource updating - %s", err)
+					return r.manageError(ctx, dbucr, err, false)
+				}
+				dbucr.Status.Created = true
+			} else {
+				dbucr.Status.Phase = DBUCR_PHASE_UPDATING
+				r.Log.Info(fmt.Sprintf("updating a user %s", dbucr.GetObjectMeta().GetName()))
+				if err := database.UpdateUser(db, dbuser, adminCred); err != nil {
+					return r.manageError(ctx, dbucr, err, false)
+				}
+			}
+			dbucr.Status.Status = true
+			dbucr.Status.Phase = DBUCR_PHASE_READY
+			dbucr.Status.DatabaseName = dbucr.Spec.DatabaseRef
+		}
+	}
+
+	return reconcileResult, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *DbUserReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&kindav1beta1.DbUser{}).
+		Complete(r)
+}
+
+func isDbUserChanged(dbucr *kindav1beta1.DbUser, userSecret *corev1.Secret) bool {
```

Comment:
possible to just remove this.

Reply:
I've added a new flag to the main.go, and a new condition here, so it's possible to choose whether it's wanted to check for changes or not